contracts/potlock/src/potlock_interactions.rs

```diff
@@ -0,0 +1,106 @@
+use crate::potlock_requirements;
+use crate::potlock_storage::{self, Pot, Project};
+use crate::potlock_storage::{PotlockId, ProjectId};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[multiversx_sc::module]
+pub trait PotlockInteractions:
+    potlock_requirements::PotlockRequirements
+    + potlock_storage::PotlockStorage
+    + multiversx_sc_modules::only_admin::OnlyAdminModule
+{
+    #[payable("*")]
+    #[endpoint(addPot)]
+    fn add_pot(&self, name: ManagedBuffer, description: ManagedBuffer) {
+        let payment_for_adding_pot = self.call_value().single_esdt();
```

Comment:
Not mandatory, but you could add these 2 storages under a single one, that keeps a EsdtTokenPayment struct. And you can check here if the payments are equal.